torch_geometric/loader/neighbor_loader.py

```diff
@@ -198,8 +205,16 @@ def __call__(self, index: Union[List[int], Tensor]):
 
         elif self.data_cls == 'custom' or issubclass(self.data_cls,
                                                      HeteroData):
-            return self._hetero_sparse_neighbor_sample(
-                {self.input_type: index}) + (index.numel(), )
+            if not self.disjoint:
+                return self._hetero_sparse_neighbor_sample(
+                    {self.input_type: index}) + (index.numel(), )
+            else:
+                result = []
+                for i in index.size(0):
```

Comment:
Does this seem like it would be very slow?

Reply:
> Yep makes sense 👍 This is what I understood from the current implementation.

Thanks for looking into it! I'll make the suggested change you made:)

I'm testing out both approaches of doing it in PyG v.s. c++ implementation in torch sparse (and later pyg-lib) to see which one could be better. Yes it does seem slower when doing it in PyG. However, this approach of separate calls to c++ could potentially benefit from easy parallelization (which is currently not utilized here).
So I'll get back to you with more insights once I can test both options out more thoroughly.